.github/workflows/security-gate.yml

```
@@ -0,0 +1,155 @@
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Run Security Scan
+        run: |
+          npm ci
+          ./security-scan.sh
+
+      - name: Security Gate Check
+        id: security-gate
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:1:39: Double quote to prevent globbing and word splitting

(shellcheck)